docs/administration-guide/named-operations.md

```diff
@@ -0,0 +1,907 @@
+                "includeIncomingOutGoing" : "OUTGOING"
+            } ]
+        },
+        "overwriteFlag" : true,
+        "readAccessRoles" : [ "read-user" ],
+        "writeAccessRoles" : [ "write-user" ]
+        }
+        ```
+
+    === "Python"
+
+        ``` python
+        g.AddNamedOperation( 
+        operation_chain=g.OperationChainDAO( 
+            operations=[ 
+            g.GetAdjacentIds( 
+                include_incoming_out_going="OUTGOING" 
+            ), 
+            g.GetAdjacentIds( 
+                include_incoming_out_going="OUTGOING" 
+            ) 
+            ] 
+        ), 
+        operation_name="2-hop", 
+        description="2 hop query", 
+        read_access_roles=[ 
+            "read-user" 
+        ], 
+        write_access_roles=[ 
+            "write-user" 
+        ], 
+        overwrite_flag=True 
+        )
+        ```
+
+Following on from this, you can then run your new NamedOperation:
+
+!!! example "Running your NamedOperation"
+
+    === "Java"
+
+        ``` java
+        final NamedOperation<EntityId, CloseableIterable<EntityId>> operation =
+        new NamedOperation.Builder<EntityId, CloseableIterable<EntityId>>()
+                .name("2-hop")
+                .input(new EntitySeed(1))
+                .build();
+        ```
+
+    === "JSON"
+
+        ``` json
+        {
+            "class" : "NamedOperation",
+            "input" : [ {
+                "class" : "EntitySeed",
+                "class" : "EntitySeed",
+                "vertex" : 1
+            } ],
+            "operationName" : "2-hop"
+        }
+        ```
+
+    === "Python"
+
+        ``` python
+        g.NamedOperation( 
+            input=[ 
+                g.EntitySeed( 
+                    vertex=1 
+                ) 
+            ], 
+            operation_name="2-hop" 
+        )
+        ```
+
+
+This produces the following result:
+```java
+EntitySeed[vertex=4]
+EntitySeed[vertex=3]
+EntitySeed[vertex=5]
+```
+
+Named Operations are able to take parameters which allow the OperationChain that is being executed to be configured.
+Parameters could be as simple as specifying `resultLimit` on a Limit Operation, they could specify a custom view to use in an Operation or even the input to an Operation.
+
+When adding a NamedOperation with parameters to an OperationChain it must be specified as a JSON string, with the parameter names enclosed in `${` and `}`. 
+For each parameter, a `ParameterDetail` object must be created which gives a description, a class type and an optional default for that parameter.
+As the default is optional you can alternatively indicate that the parameter must be provided and that there is no default.
+
+The following code adds a NamedOperation with a parameter that allows the result limit for the OperationChain to be set:
+
+!!! example "Adding a NamedOperation with parameters"
+
+    === "Java"
+
+        ``` java
+        final String opChainString = "{" +
```

Comment:
Not sure if its worth it but this seems like it could be very error prone to typos would using a `JSONObject()` and building it up be a better example e.g. something like:

```java
String opChainString = new JSONObject()
    .put("operations", new JSONArray()
        .put(new JSONObject()
            .put("class", "GetAdjacentIds")
            .put("includeIncomingOutGoing", "OUTGOING"))
        .put(new JSONObject()
            .put("class", "GetAdjacentIds")
            .put("includeIncomingOutGoing", "OUTGOING"))
        .put(new JSONObject()
            .put("class", "Limit")
            .put("resultLimit", "${param1}")))
    .toString();
```

Reply:
I agree that this would be better.